opentracing/src/main/java/opentracing/Span.java

```diff
@@ -65,29 +63,7 @@
   Span setTag(String key, Number value);
 
   /**
-   * {@code message} is a format string and can refer to fields in the payload by path, like so:
-   *
-   * <pre>{@code
-   *
-   * span.info("first transaction is worth ${transactions[0].amount} ${transactions[0].currency}",
-   *     ImmutableMap.of(
-   *       "transactions", asList(
-   *         Transaction.builder().amount(10).currency("USD").build(),
-   *         Transaction.builder().amount(11).currency("USD").build(),
-   *       )
-   *     )
-   * );
-   * }</pre>
-   *
-   * @param message {@link Formatter format string} that can refer to fields in the args payload.
-   * @param args arbitrary payload
-   */
-  // See https://github.com/opentracing/opentracing.github.io/issues/30 about parameterization
-  Span info(String message, Object... args);
-
-  /** Same as {@link #info}, but for warnings. */
-  Span warning(String message, Object... args);
-
-  /** Same as {@link #info}, but for errors. */
-  Span error(String message, Object... args);
+   * @todo
+   **/
+  Span event(String message, /* @Nullable */ Object payload);
```

Comment:
logEvent is intuitive if we consider the implicit expectation that all
things logged result in the same struct. Ex I think somewhere else in OT,
theres a log struct that has an optional field for event (in anticipation
that there will be later features which don't record event, I suppose)

We tried logEvent in Sleuth, and it hasn't raised alarm, but of course it
hasn't been released either! we chose not to add a parameter for payload,
as we have no reason to support that.

Interestingly, if you look at log4j 2, there's a logEvent op with ties in
with its propagation api. Ex people can put tags into a context used for
later formatting of the event.

https://logging.apache.org/log4j/2.0/manual/eventlogging.html

On timestamp, I would guess use long as the first arg and either specify
micros or say 'implementation decides'. Ex in log4j 2, their timestamped
message interface doesn't indicate precision, yet uses a long vs a
polarizing choice of Date types.